Assets/Scripts/Features/Garden/PlantVisual.cs

```diff
@@ -20,6 +20,7 @@ public class PlantVisual : BaseVisual<Garden>
         
         public void StartGrowing()
         {
+            StopCoroutine(GrowCoroutine());
```

Comment:
I am not sure this works well. You need to either StopCoroutine() the Coroutine object that was returned from the StartCoroutine() or call StopAllCoroutines().

Maybe this also works, but I'm skeptic because you are creating an entirely new Enumerator to cancel.

Reply:
Found the real problom :)